src/graphql/resolvers/monsterResolver.ts

```diff
@@ -1,29 +1,43 @@
-import { levelObjectToArray, resolveChoice, resolveDc, resolveUsage } from './common.js';
+import { levelObjectToArray, resolveChoice, resolveDc } from './common.js';
 
 import ConditionModel from '../../models/condition/index.js';
 import DamageTypeModel from '../../models/damageType/index.js';
 import MonsterModel from '../../models/monster/index.js';
 import ProficiencyModel from '../../models/proficiency/index.js';
 import SpellModel from '../../models/spell/index.js';
 
-const resolveDamage = async damage => {
+import {
+  Monster,
+  ActionDamage,
+  ActionUsage,
+  SpecialAbilityUsage,
+} from '../../models/monster/types';
+
+const resolveUsage = (usage: ActionUsage | SpecialAbilityUsage) => {
+  const resolvedUsage: Record<any, any> = {
+    ...usage,
+    type: usage.type.toUpperCase().replace(/\s+/g, '_'),
+  };
+  if ('rest_types' in usage)
+    resolvedUsage.rest_types = usage.rest_types?.map((rt: string) => rt.toUpperCase());
+
+  return resolvedUsage;
+};
+
+const resolveDamage = async (damage: ActionDamage[]) => {
   const damageTypes = await DamageTypeModel.find({
     index: { $in: damage.filter(d => d.damage_type).map(d => d.damage_type.index) },
   }).lean();
 
   return damage.map(async d => {
-    const newDamage = {
+    const newDamage: Record<any, any> = {
```

Comment:
It might make sense to make a type for `newDamage` somewhere in this file. Either way, I don't think it makes sense to use `any` as the key of this record.